addons/l10n_in/models/account_invoice.py

```diff
@@ -73,6 +114,48 @@ def _onchange_name_warning(self):
             }}
         return super()._onchange_name_warning()
 
+    def _l10n_in_calculate_aggregate_total(self, partner_pan, company_pan):
+        tax_aggregate_totals = {}
+        fiscal_year_end_date = fields.Date.from_string('%s-%s-%s' % (fields.Date.today().year + 1, self.company_id.fiscalyear_last_month, self.company_id.fiscalyear_last_day))
+        fiscal_year_start_date = fields.Date.add(fields.Date.subtract(fiscal_year_end_date, months=12), days=1)
+
+        company_ids = self.company_id
+        if self.company_id.parent_id:
+            company_ids |= self.company_id.parent_id
+            if self.company_id.parent_id.child_ids:
+                company_ids |= self.company_id.parent_id.mapped('child_ids')
+        if self.company_id.child_ids:
+            company_ids |= self.company_id.mapped('child_ids')
+        partner_ids = self.env['res.partner'].search([('l10n_in_pan', '=', partner_pan)])
+        moves_within_fiscal_year = self.search([
+            ('date', '>=', fiscal_year_start_date),
+            ('date', '<=', fiscal_year_end_date),
+            ('partner_id', 'in', partner_ids.ids),
+            ('company_id', 'in', company_ids.ids),
+        ])
+
+        for move in moves_within_fiscal_year:
+            accounts = move.invoice_line_ids.mapped('account_id')
+            for account in accounts.filtered(lambda account: account.l10n_in_tds_tcs_section not in tax_aggregate_totals):
+                tax_aggregate_totals[account.l10n_in_tds_tcs_section] = 0
+
+            if tax_aggregate_totals:
+                for line in move.invoice_line_ids.filtered(lambda line: line.account_id.l10n_in_tds_tcs_section not in line.tax_ids.mapped('tax_group_id')):
+                    tax_group_id = line.account_id.l10n_in_tds_tcs_section
+                    tax_aggregate_totals[tax_group_id] += line.price_total if tax_group_id.l10n_in_consider_tax == 'total_amount' else line.price_subtotal
+        return tax_aggregate_totals
+
+    def _l10n_in_calculate_per_transaction_total(self):
+        per_transection_total = {}
+        self.ensure_one()
```

Comment:
```suggestion
        self.ensure_one()
```
Make sure `ensure_one` is on the top and can't see any use of declaration `per_transection_total` here you are already declaring it on L152